### PR TITLE
[FEATURE] Base architecture for GitHub

### PR DESCRIPTION
**Pull request by delete-44**
- [x] Extracts some of the code from https://github.com/orbit-love/orbit-browser-extension/commit/c0a0fa4b7eea82442471ffd578f4d92104e44aa8
- [x] Adds architecture for creating & inserting widget on Github pages
- [x] Adds automated tests for this

This does not change any behaviour; none of the new code is run, except for by tests. However this lays the foundation for future work in implementing this setup

### CODE REVIEW

File: src/pages/githubIssueOrPullRequestPage.test.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+      window.location = {
+        ...originalLocation,
+        pathname: "/orbit-love/orbit-browser-extension/commits/main",
+      };
+
+      expect(page.detect()).toBe(false);
+    });
+  });
+
+  it("#findWidgetZones fetches only .timeline-comment elements", () => {
```
Comment (phacks) [inline src/pages/githubIssueOrPullRequestPage.test.js:47]: I am not 100% certain we’ll want to write these kind of DOM-based tests for finding elements on a page. The main failure scenario for those pieces of code is if the page structure of GitHub changes (e.g. they ship a redesign), which the test will not catch—and even more, the test will fail when we change the code to be correct.
I think that for these methods (`findWidgetZones`, `findUsername`, `findInsertionPoint`…) , we can skip the specs and rely on QA. What do you think?

**Review by phacks — APPROVED** [review]
I really like how this is shaping up 🙌 This looks all good to me—and as you mention, safe to merge even if it’s an in-progress work.

File: src/pages/githubIssueOrPullRequestPage.test.js (hunk 1 above)
Comment (delete-44) [reply src/pages/githubIssueOrPullRequestPage.test.js:47]: Hmmmm yeah, very fair. It feels a bit weird leaving it so untested but, yeah, these aren't really helping at all ^^

File: src/pages/githubIssueOrPullRequestPage.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import Page from "./page";
+
+export default class GitHubIssueOrPullRequestPage extends Page {
+  detect() {
+    const issuePageRegex = /.*\/.*\/issues?\/.*/;
+    const pullRequestPageRegex = /.*\/.*\/pulls?\/.*/;
```
[the remark this answers is not in the document]
Comment (delete-44) [reply src/pages/githubIssueOrPullRequestPage.js:6]: 😍 Was a great exercise in how easily extensible this architecture is. Super easy

**Pull request merged**